controllers/metal3machine_controller_test.go

```diff
@@ -427,7 +427,7 @@ var _ = Describe("Metal3Machine manager", func() {
 							Name:       "someothermachine",
 							Namespace:  "myns",
 							Kind:       "Metal3Machine",
-							APIVersion: infrav1.GroupVersion.String(),
+							APIVersion: infrav1.GroupVersion.Group,
```

Comment:
I don't think this is the correct way to do it. The test, if not passing shows that there is a problem in the matching function